server/src/com/google/refine/Refine.java

```diff
@@ -435,32 +447,92 @@ static private String fixWindowsUnicodePath(String path) {
             return path;
         }
         int pathSep = path.indexOf(File.separatorChar, q);
-        
+
         String goodPath = System.getenv("APPDATA");
         if (goodPath == null || goodPath.length() == 0) {
             goodPath = System.getenv("USERPROFILE");
             if (!goodPath.endsWith(File.separator)) {
                 goodPath = goodPath + File.separator;
             }
         }
-        
+
         int goodPathSep = goodPath.indexOf(File.separatorChar, q);
-        
+
         return path.substring(0, q) + goodPath.substring(q, goodPathSep) + path.substring(pathSep);
     }
-    
+
+}
+
+/**
+ * Validate the Host header of the HTTP request to see if it matches either a loopback IP
+ * address, localhost or an explicitly specified hostname. This is required to avoid DNS
+ * rebinding attacks against users running OpenRefine on their desktop computers.
+ */
+class ValidateHostHandler extends HandlerWrapper {
```

Comment:
I wonder if it would make sense to write one or two simple tests to show which hosts are allowed.
You would probably need to move this class to its own file or making it a nested class of `Refine`.

(I have to admit I wasn't aware of the fact that we had multiple top-level classes in `Refine.java`, it looks a bit confusing to me, and does not seem to be a widely recommended practice.)